docs/validate.ts

```diff
@@ -0,0 +1,92 @@
+import { fromMarkdown, Value } from 'mdast-util-from-markdown';
+import { mdxFromMarkdown } from 'mdast-util-mdx';
+import { mdx } from 'micromark-extension-mdx';
+import { Node, Parent } from 'mdast';
+import { MdxJsxAttribute, MdxJsxFlowElement, MdxJsxTextElement } from 'mdast-util-mdx';
+
+// A type guard to check if a node has children
+function isParent(node: Node | Parent): node is Parent {
+  return 'children' in node;
+}
+
+// Pure function to return an array of all nodes in the tree
+function flattenTree(node: Node | Parent): (Node | Parent)[] {
+  const nodes = [node];
+  if (isParent(node)) {
+    return nodes.concat(node.children.flatMap(flattenTree));
+  }
+  return nodes;
+}
+
+// Pure predicate function to check if node is a TutorialFallback element
+function isTutorialFallbackNode(node: Node): node is MdxJsxFlowElement | MdxJsxTextElement {
+  const isFlowElement = node.type === 'mdxJsxFlowElement';
+  const isJsxTextElement = node.type === 'mdxJsxTextElement';
+  const isNameTutorialFallback = 'name' in node && node.name === 'TutorialFallback';
+  
+  return (isFlowElement || isJsxTextElement) && isNameTutorialFallback;
+}
+
+// Pure function to extract filename from node attributes
+function extractFilename(node: MdxJsxFlowElement | MdxJsxTextElement): string | null {
+  const fileNameAttr = node.attributes?.find(
+    attr => attr.type === 'mdxJsxAttribute' && attr.name === 'filename'
+  ) as MdxJsxAttribute | undefined;
+  
+  return (fileNameAttr?.value as string) || null;
+}
+
+export function extractArgumentFileNames(mdxContent: Value): (string | null)[] {
+  const tree = fromMarkdown(mdxContent, {
+    extensions: [mdx()],
+    mdastExtensions: [mdxFromMarkdown()]
+  });
+
+  return flattenTree(tree)
+    .filter(isTutorialFallbackNode)
+    .map(extractFilename);
+}
+
+function validateResourceAccess(mdxContent: Value, validFilenames: string[]): boolean {
+  const argFilenames = extractArgumentFileNames(mdxContent);
+  const validResources = argFilenames.map(argFileName =>
+    validFilenames.includes(argFileName as string)
+  );
+
+  validResources.forEach((isValid, index) => {
+    if (!isValid) {
+      console.log(`\tERROR: NON-EXISTENT STATIC RESOURCE:\t "${argFilenames[index]}"`);
+    }
+  });
+
+  return validResources.every(isValid => isValid);
+}
```

Comment:
In FP, we will try to keep as many functions as pure as possible, and only do IO at the program boundaries.

So this function shouldn't do a console.log and return a boolean, instead, it should return a list of usage of non-existent resources.

Keeping a function pure makes it better for testing and debugging.